includes/discovery/applications.inc.php

```diff
@@ -67,25 +69,23 @@
     return $result;
 }, [[], []]);
 
+// enable observer for printing changes
+ModuleModelObserver::observe(\App\Models\Application::class);
+
 // Enable applications
 $current_apps = [];
 foreach ($results as $extend => $result) {
     if (isset($applications[$extend])) {
         $app = $applications[$extend];
         $current_apps[] = $app;
 
-        if (in_array($app, $enabled_apps)) {
-            echo '.';
-        } else {
-            dbInsert([
-                'device_id' => $device['device_id'],
-                'app_type' => $app,
-                'discovered' => 1,
-                'app_status' => '',
-                'app_instance' => '',
-            ], 'applications');
-
-            echo '+';
+        if (! in_array($app, $enabled_apps)) {
+            $app_obj = Application::withTrashed()->firstOrNew(['device_id' => $device['device_id'], 'app_type' => $app]);
+            if ($app_obj->trashed()) {
+                $app_obj->restore();
```

Comment:
ModuleModelObserver does not currently handle the restored event, so this won't output anything.

Reply:
Ahh! Okay. You mean it does not hit on the restored bit.

 Added restored to ModuleModelObserver.